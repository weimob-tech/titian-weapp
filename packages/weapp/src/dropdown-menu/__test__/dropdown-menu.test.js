const simulate = require('miniprogram-simulate');
const { getComponent } = require('../../common/test/getComponent');

describe('test component', () => {
  jest.setTimeout(10000);
  it('should render success', (done) => {
    const comp = getComponent('dropdown-menu');
    const child = getComponent('dropdown-item', {
      title: 'title'
    });

    comp.triggerLifeTime('attached');
    child.triggerLifeTime('attached');

    const getParentRelationNodes = comp.instance.getRelationNodes;
    const getChildRelationNodes = child.instance.getRelationNodes;
    const parentAnimate = comp.instance.animate;

    comp.instance.animate = jest.fn();
    comp.instance.getRelationNodes = jest.fn(() => [child.instance]);
    child.instance.getRelationNodes = jest.fn(() => [comp.instance]);

    Promise.resolve()
      .then(() => {
        comp.instance.updateChildrenData();
        expect(
          comp.querySelector('.titian-dropdown-menu >>> .titian-button').dom.innerHTML.indexOf('title') > -1
        ).toBeTruthy();
        expect(comp).toMatchSnapshot();

        comp.querySelector('.titian-dropdown-menu >>> .titian-button').dispatchEvent('tap');
      })
      .then(async () => {
        await simulate.sleep(400);
        expect(child.data.visible).toBeTruthy();
        comp.querySelector('.titian-dropdown-menu >>> .titian-button').dispatchEvent('tap');
      })
      .then(async () => {
        await simulate.sleep(400);
        expect(child.data.visible).toBeFalsy();
      })
      .finally(() => {
        comp.instance.getRelationNodes = getParentRelationNodes;
        child.instance.getChildRelationNodes = getChildRelationNodes;
        comp.instance.animate = parentAnimate;
        child.triggerLifeTime('detached');
        done();
      });
  });

  it('should render props success', () => {
    const comp = getComponent('dropdown-menu', {
      icon: 'icon-test',
      mode: 'multiple',
      direction: 'up',
      activeColor: '#f00',
      hasMask: false,
      closeOnMask: true
    });
    const child = getComponent('dropdown-item');

    comp.triggerLifeTime('attached');
    child.triggerLifeTime('attached');

    const getParentRelationNodes = comp.instance.getRelationNodes;
    const getChildRelationNodes = child.instance.getRelationNodes;
    const parentAnimate = comp.instance.animate;

    comp.instance.animate = jest.fn();
    comp.instance.getRelationNodes = jest.fn(() => [child.instance]);
    child.instance.getRelationNodes = jest.fn(() => [comp.instance]);

    Promise.resolve()
      .then(() => {
        comp.instance.updateChildrenData();
      })
      .then(() => {
        expect(child.data.mode).toBe('multiple');
        expect(child.data.direction).toBe('up');
        expect(child.data.activeColor).toBe('#f00');
        expect(child.data.icon).toBe('icon-test');
        expect(child.data.hasMask).toBeFalsy();
        expect(child.data.closeOnMask).toBeTruthy();
      })
      .finally(() => {
        comp.instance.getRelationNodes = getParentRelationNodes;
        child.instance.getChildRelationNodes = getChildRelationNodes;
        comp.instance.animate = parentAnimate;
      });
  });
});
