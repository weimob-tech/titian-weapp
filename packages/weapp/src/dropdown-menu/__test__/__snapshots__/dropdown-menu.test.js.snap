// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="ext-class titian-dropdown-menu"
    style=""
  >
    <wx-view
      class="titian-dropdown-menu-title-wrap"
    >
      <wx-view
        class="title-class titian-dropdown-menu-title titian-dropdown-menu-title-0"
        data-idx="{{0}}"
      >
        <ti-button
          data-idx="{{0}}"
          bind:click="onTitleToggle"
        >
          <wx-button
            appParameter=""
            class="ext-class titian-button-block titian-button-hairline titian-button titian-button-text titian-button-type-primary titian-button-size-big"
            data-detail="{{null}}"
            disabled="{{false}}"
            formType=""
            hoverClass="active"
            hoverStartTime="{{20}}"
            hoverStayTime="{{70}}"
            id="titian-button"
            lang="en"
            loading="{{false}}"
            openType=""
            scope=""
            sendMessageImg=""
            sendMessagePath=""
            sendMessageTitle=""
            sessionFrom=""
            showMessageCard="{{false}}"
            size="big"
            style="--text-color:var(--button-text-color,black);--loading-color:var(--button-loading-color, --text-color);--background-color:transparent;--active-color:transparent"
            bind:FollowLifestyle="bindFollowLifestyle"
            bind:contact="onContact"
            bind:error="onError"
            bind:getAuthorize="bindGetAuthorize"
            bind:getphonenumber="onGetphonenumber"
            bind:getuserinfo="onGetuserinfo"
            bind:launchapp="onLaunchapp"
            bind:opensetting="onOpensetting"
            bind:tap="onClick"
          >
            <wx-view
              class="titian-button-content button-inner-class"
            >
              
        title
        
              <wx-view
                class="titian-dropdown-menu-icon"
              >
                <ti-icon>
                  <wx-view
                    class="titian-icon-base titian-icon titian-icon-arrow-down ext-class"
                    style=""
                    bind:tap="onClick"
                  />
                </ti-icon>
              </wx-view>
            </wx-view>
          </wx-button>
        </ti-button>
      </wx-view>
    </wx-view>
  </wx-view>
</main>
`;
